infra/modules/aws-psoxy-bulk/main.tf

```diff
@@ -3,7 +3,7 @@ terraform {
     # for the infra that will host Psoxy instances
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 4.12"
+      version = "~> 4.29"
```

Comment:
In `4.12` [version](https://registry.terraform.io/providers/hashicorp/aws/4.12.1/docs/resources/ssm_parameter), `ssm_parameter` requires to have `value` as mandatory; meanwhile in newer versions is mandatory to have or `value` or  `insecure_value` -> https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ssm_parameter

BTW, this failed to me because I had a older version of this AWS provider module in my local. Not an issue if you are going to run it from scratch, but it we update the code without changing the provider in case the function we modify is not included in the older ones, someone could face with an issue here.

I know that TF recommends to have its module with its own provider [requirements](https://www.terraform.io/language/modules/develop/providers#provider-version-constraints-in-modules), something that Googe doesn't [recommend](https://cloud.google.com/docs/terraform/best-practices-for-terraform#providers-backends). 

I'd say that I prefer the Google recommendations, having a single provider in the root module we will enforce us to have each module always updated with that version and to avoid issues with previously installed module, not sure if is interesting to do next time we generate new module tags

Reply:
i'm OK with upgrading the AWS module; i think it should be upgraded.

TF is thinking of each module being really decoupled and re-usable ... I think that's *eventually* they way we'll go, but so long as all in mono-repo, I think OK at the top.